src/frontend/components/DashboardVideoLiveControlPane/widgets/DashboardVideoLiveWidgetGeneralTitle/index.spec.tsx

```diff
@@ -0,0 +1,292 @@
+    });
+
+    await act(async () => userEvent.click(recordingToggleButton));
+    expect(fetchMock.calls()).toHaveLength(1);
+    expect(fetchMock.lastCall()![0]).toEqual('/api/videos/video_id/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'PATCH',
+      body: JSON.stringify({
+        allow_recording: true,
+      }),
+    });
+    expect(report).not.toHaveBeenCalled();
+  });
+
+  it('clicks on the toggle button for deactivate the live recording', async () => {
+    const mockedVideo = videoMockFactory({
+      id: 'video_id',
+      allow_recording: true,
+    });
+
+    fetchMock.patch('/api/videos/video_id/', {
+      ...mockedVideo,
+      allow_recording: false,
+    });
+
+    const queryClient = new QueryClient();
+
+    render(
+      wrapInIntlProvider(
+        <QueryClientProvider client={queryClient}>
+          <DashboardVideoLiveWidgetGeneralTitle video={mockedVideo} />,
+        </QueryClientProvider>,
+      ),
+    );
+
+    screen.getByText('Activate live recording');
+    const recordingToggleButton = screen.getByRole('checkbox', {
+      name: 'Activate live recording',
+    });
+
+    await act(async () => userEvent.click(recordingToggleButton));
+    expect(fetchMock.calls()).toHaveLength(1);
+    expect(fetchMock.lastCall()![0]).toEqual('/api/videos/video_id/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'PATCH',
+      body: JSON.stringify({
+        allow_recording: false,
+      }),
+    });
+    expect(report).not.toHaveBeenCalled();
+  });
+
+  it('clicks on the toggle button for activate live recording, but backend returns an error', async () => {
+    const mockedVideo = videoMockFactory({
+      id: 'video_id',
+      allow_recording: false,
+    });
+
+    fetchMock.patch('/api/videos/video_id/', 500);
+
+    const queryClient = new QueryClient();
+
+    render(
+      wrapInIntlProvider(
+        <QueryClientProvider client={queryClient}>
+          <Toaster />
+          <DashboardVideoLiveWidgetGeneralTitle video={mockedVideo} />,
+        </QueryClientProvider>,
+      ),
+    );
+
+    screen.getByText('Activate live recording');
+
+    const recordingToggleButton = screen.getByRole('checkbox', {
+      name: 'Activate live recording',
+    });
+    await act(async () => userEvent.click(recordingToggleButton));
+
+    expect(fetchMock.calls()).toHaveLength(1);
+    expect(fetchMock.lastCall()![0]).toEqual('/api/videos/video_id/');
+    expect(fetchMock.lastCall()![1]).toEqual({
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      method: 'PATCH',
+      body: JSON.stringify({
+        allow_recording: true,
+      }),
+    });
+    screen.getAllByText('Recording activation status update has failed !');
+  });
+});
+
+describe('test text input in <DashboardVideoLiveWidgetGeneralTitle />', () => {
```

Comment:
All test in this describe are really slow. You have to investigate why, they took 20 seconds to execute on my laptop

Reply:
And I think you should mock the timer here, maybe it is the debounce that is slowing down your tests.

Reply:
Taking advantage of jest's fake timers was useful to make debounce end faster and reduce time's test. But the biggest problem was `react-hot-toast` library, which is not easily testable and needed a hack to be testable :  
[7a30b9f3854](https://github.com/openfun/marsha/pull/1503/commits/7a30b9f3854aeff9915977940c8084b5f212f391)